src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtil.java

```diff
@@ -281,7 +281,14 @@ public static boolean isTypeDeclaration(int type) {
      * @return true if type matches one of the given types.
      */
     public static boolean isOfType(int type, int... types) {
-        return Arrays.stream(types).anyMatch(tokenType -> tokenType == type);
+        boolean matching = false;
```

Comment:
We can go further and add overloads with one or two int parameters. Like this
```java
public static boolean isOfType(int type, int first, int second) {
    return type == first || type == second;
}
```
The JIT compiler will definitely inline such methods .

Reply:
> We can go further and add overloads with one or two int parameters. Like this
> 
> ```java
> public static boolean isOfType(int type, int first, int second) {
>     return type == first || type == second;
> }
> ```
> 
> The JIT compiler will definitely inline such methods .

Interesting idea, but are there no usages where `types` is constructed dynamically? Then I think this would not work.

With regards to performance, I would be curious to see how much the difference is, but my expectation is it to not be a major difference from the native loop. With regards to maintainability I do feel like this takes a major hit, as you'd need one such method for each length of `types`, but then I am unsure where exactly the performance / maintainability trade-off of Checkstyle lies.